### PR TITLE
Fix/portal module source order

### LINKED ISSUE #2323: admin: Reorder authentication sources on portal modules
It is possible to order the authentication source of a portal module by editing the configuration file manually, but if you then go in the portal module configuration page, the authentication source field is populated in a wrong way which will change the order manually configured directly in the file.

Admin field should be a sorted field

### PR DESCRIPTION
**Pull request by julsemaan**
## Need to discuss whether or not we backport this in maintenance. If so, it should be tested thoroughly given its high impact on the captive portal

# Description
Ensure that authentication sources are properly ordered for portal modules when using the admin interface

# Impacts
Portal modules / the whole captive portal

# Issue
fixes #2323 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* Fix authentication sources ordering issue for portal modules when using the administration interface (#2323)

### CODE REVIEW

**Comment by lzammit** [comment]
May 17 13:13:49 unicorn httpd_admin: May 17 13:13:49 httpd.admin(8934) ERROR: [mac:unknown] Caught exception in pfappserver::Controller::Config::PortalModule->create "Can't use an undefined value as an ARRAY reference at /usr/local/pf/lib/pf/ConfigStore/PortalModule.pm line 62." (pfappserver::PacketFence::Controller::Root::end)

**Comment by julsemaan** [comment]
@lzammit, thats when doing creating which type of portal module ?

**Comment by julsemaan** [comment]
should be fixaide

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm [hunk 1]
```diff
@@ -14,17 +14,22 @@ use HTML::FormHandler::Moose::Role;
...
 use pf::log;
+use pf::constants;
 
 after 'setup' => sub {
     my ($self) = @_;
 
     if($self->for_module->does('captiveportal::Role::MultiSource')){
-        $self->field('source_id')->multiple(1);
-        $self->field('source_id')->options([$self->options_sources(multiple => 1)]);
+        $self->field('multi_source_ids.contains')->options([$self->options_sources(multiple => 1)]);
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: $TRUE !

Comment (dwlfrth) [reply html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: Isn't 'multi_source_ids.contains' a "strange" name for a field ?
I mean, the dot

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm [hunk 2]
```diff
@@ -14,17 +14,22 @@ use HTML::FormHandler::Moose::Role;
...
+        $self->field('multi_source_ids')->inactive($FALSE);
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: Shouldn't it be the "reverse" logic ?
'active($TRUE)' is much better than 'inactive($FALSE)'

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm [hunk 3]
```diff
@@ -14,17 +14,22 @@ use HTML::FormHandler::Moose::Role;
...
     }
     else {
         $self->field('source_id')->options([$self->options_sources(multiple => 0)]);
+
+        # The multi-source field should be set to inactive so it doesn't display
+        $self->field('multi_source_ids')->inactive($TRUE);
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: Shouldn't it be the "reverse" logic ?
'active($FALSE)' is much better than 'inactive($TRUE)'

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm [hunk 4]
```diff
@@ -37,6 +42,32 @@ after 'process' => sub {
...
+sub source_fields {
+    my ($self) = @_;
+    # No need to add the multi_source_ids fields as it will be taken when looking at all the dynamic tables
+    return $self->for_module->does('captiveportal::Role::MultiSource') ? qw() : qw(source_id);
+}
+
+has_field 'multi_source_ids' =>
+  (
+    'type' => 'DynamicTable',
+    'sortable' => 1,
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: $TRUE

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm [hunk 5]
```diff
@@ -37,6 +42,32 @@ after 'process' => sub {
...
+    'do_label' => 0,
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: $FALSE

**Comment by lzammit** [comment]
Fix-aide 🍖

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm (hunk 1 above)
Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: That's the way the «framework» for ordered selects works and its hardcoded to required a field called .contains from what I know

File: html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm (hunk 2 above)
Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/Base/Form/Role/WithSource.pm]: Tell that to the formhandler team :P 
http://search.cpan.org/dist/HTML-FormHandler-0.40017/lib/HTML/FormHandler/Field.pm#Field_data

**Comment by julsemaan** [comment]
fixaide except for inactive->active since its a formhandler field attribute

**Comment by dwlfrth** [comment]
Right, I forgot you were using the "framework"

**Comment by dwlfrth** [comment]
LGTM and @lzammit said it is working more than expected

**Pull request merged**